pkg/controller/buildrun/runtime_image.go

```diff
@@ -0,0 +1,194 @@
+package buildrun
+
+import (
+	"bytes"
+	"fmt"
+	"path"
+	"strconv"
+	"strings"
+	"text/template"
+
+	buildv1alpha1 "github.com/redhat-developer/build/pkg/apis/build/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	// runtimeDockerfileTmpl Dockerfile template to be used with runtime-image, it uses Build attributes
+	// directly as template input.
+	runtimeDockerfileTmpl = `FROM {{ .Spec.Output.ImageURL }} as builder
+
+FROM {{ .Spec.Runtime.Base.ImageURL }}
+
+{{- range $k, $v := .Spec.Runtime.Env }}
+ENV {{ $k }}="{{ $v }}"
+{{- end }}
+
+{{- range $k, $v := .Spec.Runtime.Labels }}
+LABEL {{ $k }}="{{ $v }}"
+{{- end }}
+
+{{- range $dir := .Spec.Runtime.Directories }}
+{{- $parts := splitDirectories $dir }}
+COPY --from=builder "{{ index $parts 0 }}" "{{ index $parts 1 }}"
+{{- end }}
+
+{{- if .Spec.Runtime.WorkDir }}
+WORKDIR "{{ .Spec.Runtime.WorkDir }}"
+{{- end }}
+
+{{- if .Spec.Runtime.Entrypoint }}
+ENTRYPOINT [ {{ renderEntrypoint .Spec.Runtime.Entrypoint }} ]
+{{- end -}}
+`
+
+	// workspaceDir common workspace directory, where the source code is located.
+	workspaceDir = "/workspace/source"
+
+	// kanikoStable stable buildah image.
+	kanikoStable = "gcr.io/kaniko-project/executor:v0.23.0"
```

Comment:
Please make the Kaniko image configurable using the `cfg` package and some environment variable. We will probably use a custom one.

Reply:
Please consider latest commit.